### PR TITLE
Ability to present multiple maps on screen.

### PR DESCRIPTION
**Pull request by BinarySoftware**
### Pull Request Description
Enables our maps to present them in multiple instances.
Zrzut ekranu 2020-11-20 o 20 39 32

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [x] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [ ] All code has been manually tested in the "debug/interface" scene.
- [ ] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js [hunk 1]
```diff
@@ -74,11 +75,12 @@ class MapViewVisualization extends Visualization {
...
             this.dom.removeChild(this.dom.lastChild);
         }
-
+        MAP_ID += 1;
+        const id      = "map_" + MAP_ID;
         const width   = this.dom.getAttributeNS(null,"width");
         const height  = this.dom.getAttributeNS(null,"height");
         const mapElem = document.createElement("div");
-        mapElem.setAttributeNS(null,"id"   ,"map");
+        mapElem.setAttributeNS(null,"id"   ,id);
```
Comment (wdanilo) [inline src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js:83]: Are you sure you want to change the ID on EVERY time the map gets data? This is inside of function `onDataReceived`. I feel its not the right thing to do

**Review by wdanilo — CHANGES_REQUESTED** [review]

File: src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js (hunk 1 above)
Comment (BinarySoftware) [reply src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js]: fixed

File: src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js [hunk 2]
```diff
@@ -69,16 +70,23 @@ document.head.appendChild(styleHead);
...
 
     onDataReceived(data) {
         while (this.dom.firstChild) {
             this.dom.removeChild(this.dom.lastChild);
         }
 
+        if (this.mapId === 0) {
+            MAP_ID     += 1;
+            this.mapId  = MAP_ID;
+        }
```
Comment (wdanilo) [inline src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js:83]: Create utility function for that next to global variable, make section for it and add documentation. Implementation of the incremenation of the global variable should not be part fo the loigic in this visualization.

File: src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js [hunk 3]
```diff
@@ -111,7 +119,7 @@ class MapViewVisualization extends Visualization {
         let controller = ok(parsedData.controller) ? parsedData.controller : true;
 
         const deckgl = new deck.DeckGL({
-            container: 'map',
+            container: id,
             mapboxApiAccessToken: TOKEN,
             mapStyle,
             initialViewState: {longitude,latitude,zoom,pitch},
```
Comment (wdanilo) [inline src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js:125]: This is very bad. It even can cause our GUI to crash and needs to be fixed ASAP. We should never reinitialize visualizations on new incomming data!

**Review by wdanilo — CHANGES_REQUESTED** [review]

**Comment by wdanilo** [comment]
Moreover, there is another serious issue here that we need to fix ASAP as it crashes our GUI. Whenever visualization is closed, the WebGL context of deck.gl needs to be destroyed - the DOM elements need to be remmoved etc. Let's investigate that as soon as the above comments will be fixed.

**Comment by MichaelMauderer** [comment]
Closed in favour of  #990

**Pull request closed without merging**